### ISSUE TITLE
How to clone linked pages?

### ISSUE THREAD

**Issue by rstmsn** [rajatomar788/pywebcopy#63]
I'm running the following example. The target page downloads OK, however none of the linked pages are being downloaded. Is there a configuration flag I can set, to download hyperlinked pages?

`from pywebcopy import save_website

kwargs = {'project_name': 'xxxx-clone-eb'}

save_website(
    url='https://xxxxx.com/ARTICLES/xxxx.htm',
    project_folder='/Users/xxxx/Documents/Code/xxxxx/EB',
    **kwargs
)
`

**Comment by rajatomar788** [comment]
Are the hyperlinked pages hosted on the same site domain or outside?

**Comment by rstmsn** [comment]
On the same domain. Currently, it doesn't seem to be following any hyperlinks. In my project_folder, i'm only seeing 1 .html file, despite the fact that this page contains many linked pages.

**Comment by rajatomar788** [comment]
The pywebcopy builds a hierarchical structure meaning your hyperlinked pages might be in some folders relative to the main html file.

**Comment by rstmsn** [comment]
No. There is only 1 .html file, no other folders, no other files. When clicking a hyperlink within the .html file, it 404s because the package has not followed / downloaded any of the hyperlinks. Why would this be?

**Comment by unmurphy** [comment]
facing the same issue, any updates?

**Comment by rajatomar788** [comment]
It could be server side site specific issue. Maybe the hyperlinks are not resolving properly due to bad url or html formatting.

**Comment by rstmsn** [comment]
> It could be server side site specific issue. Maybe the hyperlinks are not resolving properly due to bad url or html formatting.

server side is fine. I just successfully cloned the same site manually, using wget. For others who might benefit from this code-free solution:

`wget --recursive --no-clobber --page-requisites --html-extension --convert-links --restrict-file-names=windows --domains example.org --no-parent www.example.org/directory/`

**Comment by unmurphy** [comment]
@rstmsn for your solution, i found each of html <img> src  property was still using the origin resource. do you have any other idea?

**Comment by rstmsn** [comment]
> @rstmsn for your solution, i found each of html src property was still using the origin resource. do you have any other idea?

recursive find & replace using grep?

**Comment by monim67** [comment]
Any update on this? Facing the same issue.

**Comment by rajatomar788** [comment]
Ok just use 'save_website' function instead of save_webpage

**Comment by monim67** [comment]
> Ok just use 'save_website' function instead of save_webpage

The issue is with the `save_website` function itself. It's downloading a single page just like `save_webpage`.

**Comment by BrandonKMLee** [comment]
Does anyone here knows how to crawl a whole subdomain? Currently trying to test something out.

**Comment by rajatomar788** [comment]
@BrandonKMLee you can modify the session object which is created in the save_website function to discard any unwanted domains. You should see the source code of save_website function and then just replace the default session with a modified one.

**Comment by BrandonKMLee** [comment]
Let's say there are these three scenarios:
1. You want to scrape "'https://www.nateliason.com/notes*" but nothing else around "https://www.nateliason.com", and that all child URLs of "'https://www.nateliason.com/notes/{pages}" are traceable to the parent page
2. You want to scrape the articles within "https://paulminors.com/resources/book-summaries" but the URLs are shortened in there, and the site also have other unrelated articles
3. You want to scrape the whole website (any) but not any other domain

I am trying to figure out which value to change within `Session` since dunno how it is tied to `save_website`
